### ISSUE TITLE
Confused of your inference mode

### ISSUE THREAD

**Issue by qijiezhao** [NVIDIA/flownet2-pytorch#5]
Sorry to impose, I was confused when I was trying to extract flow images from my custom dataset(or just a few frames), However, after I figured out the details of your code, only limited datasets are supported?  It's not very friendly to use. 
Would you plan to write a test.py as a friendly-use api example？

**Comment by fitsumreda** [comment]
Hi @qijiezhao , `ImagesFromFolder` is the dataloader for regular images for simplistic test purpose, it expects two input arguments
`root - folder that contains sequentially names RGB images`
`iext - an extension for the image files`
both of these can be passed in as arguments, `inference_dataset_root` and `inference_dataset_png`.
Could you explain a little about your custom dataset? 
Any way `ImagesFromFolder` can be extended?

**Comment by qijiezhao** [comment]
Hi @fitsumreda , Thx for your reply, here I modified some code in main.py and it is approaching worked out:
     if exists(args.inference_dataset_root):
            #inference_dataset = args.inference_dataset_class(args, False, **tools.kwargs_from_args(args, 'inference_dataset'))
            #block.log('Inference Dataset: {}'.format(args.inference_dataset))
            #block.log('Inference Input: {}'.format(' '.join([str([d for d in x.size()]) for x in inference_dataset[0][0]])))
            #block.log('Inference Targets: {}'.format(' '.join([str([d for d in x.size()]) for x in inference_dataset[0][1]])))
            #inference_loader = DataLoader(inference_dataset, batch_size=args.inference_batch_size, shuffle=False, **gpuargs)
            inference_loader = DataLoader(
                datasets.ImagesFromFolder(args=args,is_cropped=False,root=args.inference_dataset_root,iext='jpg'),
                batch_size=args.inference_batch_size,shuffle=False,**gpuargs
            )
Now, without checkpoints file,  I am working about this.

**Comment by sklf** [comment]
@fitsumreda @qijiezhao Hello, I read your discussion. And I used the code as above. It is as follows:
```
if exists(args.inference_dataset_root):
        inference_loader = DataLoader(datasets.ImagesFromFolder(args=args, is_cropped=False, root=args.inference_dataset_root, iext='png'),
            batch_size =args.effective_inference_batch_size, shuffle=False, **inf_gpuargs)
```
and my custom dataset has 100 '*.png' images. However, I ran the model Flownet2S with the following code 
```
python main.py --inference --model FlowNet2S --save_flow --inference_dataset_root ~/files/Dataset/posetrack/Flowimgs/imgs/val/000342_mpii_test/ --resume ./work/FlowNet2-S_checkpoint.pth.tar 
```
and found I only got 24 flo file. I used 3 GPUs to ran it. So can you tell me why I can't get all flo files? Thanks!
